actions/datadog-release-metric/DatadogReleaseMetric.ts

```diff
@@ -0,0 +1,90 @@
+import * as core from "@actions/core";
+import { Context } from "@actions/github/lib/context";
+import {
+  MetricsApi,
+  MetricsApiSubmitMetricsRequest,
+} from "@datadog/datadog-api-client/dist/packages/datadog-api-client-v1";
+
+const METRIC_TYPE = "count";
+const METRIC_NAME = "library.release.count";
+const REF_REGEX = /^refs\/(.+?)\/(.+)$/;
+
+export interface MetricParams {
+  type: string;
+  name: string;
+  value: number;
+  tags: string[];
+}
+
+export default class DatadogReleaseMetric {
+  private readonly type: string = METRIC_TYPE;
+  private readonly name: string = METRIC_NAME;
+
+  constructor(private readonly context: Context, private datadog: MetricsApi) {}
+
+  async run() {
+    const params: MetricParams = {
+      type: this.getType(),
+      name: this.getName(),
+      tags: this.getTags(this.getOrg(), this.getRepo(), this.getVersion()),
+      value: 1,
+    };
+    await this.sendMetric(params);
+  }
+
+  getVersion(): string {
+    const [ref, refType, refName] = this.context.ref.match(REF_REGEX) || [];
+
+    if (!ref) {
+      throw new Error(`Invalid ref: ${this.context.ref}`);
+    }
+    if (refType !== "tags") {
+      throw new Error(`Invalid ref type, must be "tags": ${refType}`);
+    }
+
+    return refName;
+  }
+
+  getTags(owner: string, repo: string, version: string): string[] {
+    return [
+      `org:${owner}`,
+      `repo:${repo}`,
+      "type:helper",
+      `version:${version}`,
```

Comment:
Don't know how valuable this last tag is. It's not something that we can correlate (not sure if that's the right word) across repos or languages.

Reply:
I  thought it'd be good "metadata" to have associated with a release datapoint, not necessarily something we would filter against, but I agree it's not strictly required. We can always add it later or cross-check with GitHub. I'll tear it out